src/host/screenInfo.cpp

```diff
@@ -2309,6 +2310,11 @@ OutputCellRect SCREEN_INFORMATION::ReadRect(const Viewport viewport) const
 // - will throw exception on error.
 OutputCellIterator SCREEN_INFORMATION::Write(const OutputCellIterator it)
 {
+    __assume(!_insertMode);

```

Comment:
**NOTE**: This did nothing in my testing, but it makes me feel better. The optimizer actually deleted this function _and_ `TextBuffer::Write` (and `...::Insert` (!)) because it saw right through our trick. It lifted the boolean load from `SCREEN_INFORMATION::_insertMode` directly into `WriteCharsLegacy` and passed it down into the fully-realized `ROW::Write(..., insert)` directly (!)